app/actions/views/more_dms.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {getDirectChannelName} from 'service/utils/channel_utils';
+import {createDirectChannel} from 'service/actions/channels';
+import {getTeamMember} from 'service/actions/teams';
+import {handleSelectChannel, toggleDMChannel} from 'app/actions/views/channel';
+
+export function makeDirectChannel(otherUserId) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {currentId} = state.entities.users;
+        const channelName = getDirectChannelName(currentId, otherUserId);
+        const {channels, myMembers} = state.entities.channels;
+        const channel = Object.values(channels).find((c) => c.name === channelName);
+        const teamId = state.entities.teams.currentId;
+
+        await getTeamMember(teamId, otherUserId)(dispatch, getState);
+
+        if (channel && myMembers[channel.id]) {
+            toggleDMChannel(otherUserId, 'true')(dispatch, getState).
+            then(() => {
```

Comment:
I'm definitely guilty of this, but just for consistency going forward if we're marking a function as async do we want to go ahead and use await for all promises within? Seems strange to mix the two.